src/api/hooks/useNotesOfUser.ts

```diff
@@ -0,0 +1,51 @@
+import { useCallback } from 'react';
+import {
+  InfiniteData,
+  UseInfiniteQueryResult,
+  useInfiniteQuery,
+} from '@tanstack/react-query';
+
+import { fetchNotesOfUser, QUERY_KEYS } from 'api';
+import { RESPONSE_LIMIT } from 'api/endpoints';
+import { filterNotes } from 'api/mappers';
+import { FILTER_OPTIONS } from 'config';
+import { INote } from 'types';
+
+export const useNotesOfUser = (
+  userId: string,
+  filterOption?: FILTER_OPTIONS,
+  filterValue?: string,
+) => {
+  const {
+    data: notesResponse,
+    hasNextPage,
+    hasPreviousPage,
+    isLoading,
+    isFetching,
+    fetchNextPage,
+  }: UseInfiniteQueryResult<INote[], Error> = useInfiniteQuery({
+    queryKey: [QUERY_KEYS.NOTES],
+    queryFn: ({ pageParam = 1 }) => fetchNotesOfUser(userId, pageParam),
+    getNextPageParam: (lastPage, allPages) => {
+      const nextPage =
+        lastPage.length === RESPONSE_LIMIT ? allPages.length + 1 : undefined;
```

Comment:
```suggestion
      const nextPage =
        lastPage.length === RESPONSE_LIMIT &&  allPages.length + 1;
```

Reply:
you should remove magic number, also please notice, that it's a bad practice to use undefined in such cases
undefined cannot be set 

and also suggested change is preferred, cause it do the same